server/routes/shared/action-plan/actionPlanProgressView.ts

```diff
@@ -0,0 +1,87 @@
+import { TableArgs, TableArgsCell, TagArgs } from '../../../utils/govukFrontendTypes'
+import ActionPlanProgressPresenter from './actionPlanProgressPresenter'
+import ActionPlanUtils, { ActionPlanStatus } from '../../../utils/actionPlanUtils'
+import DateUtils from '../../../utils/dateUtils'
+import ViewUtils from '../../../utils/viewUtils'
+
+export default class ActionPlanProgressView {
+  constructor(private readonly presenter: ActionPlanProgressPresenter) {}
+
+  tableArgs(tagMacro: (args: TagArgs) => string, csrfToken?: string): TableArgs {
+    const rows: TableArgsCell[][] = []
+
+    if (this.presenter.includeCurrentActionPlanRow) {
+      rows.push(this.getCurrentActionPlanRow(tagMacro, csrfToken))
+    }
+
+    rows.push(
+      ...this.presenter.approvedActionPlanSummaries.map((summary, i) => [
+        {
+          text: (this.presenter.approvedActionPlanSummaries.length - i).toString(),
+          classes: 'action-plan-version',
+        },
+        {
+          text: tagMacro(ActionPlanUtils.actionPlanTagArgs(ActionPlanStatus.Approved)),
+          classes: 'action-plan-status',
+        },
+        {
+          text: DateUtils.formattedDate(summary.submittedAt),
+          classes: 'action-plan-submitted-date',
+        },
+        {
+          text: DateUtils.formattedDate(summary.approvedAt),
+          classes: 'action-plan-approved-date',
+        },
+        {
+          html: `<a href="/${this.presenter.userType}/action-plan/${summary.id}" class="govuk-link">View action plan</a>`,
+        },
+      ])
+    )
+
+    return {
+      head: this.presenter.tableHeaders.map(header => {
+        return { text: header }
+      }),
+      rows,
+    }
+  }
+
+  private getCurrentActionPlanRow(tagMacro: (args: TagArgs) => string, csrfToken?: string): TableArgsCell[] {
+    const currentActionPlanSummary = this.presenter.currentActionPlanSummaryPresenter
+    const row: TableArgsCell[] = [
+      {
+        text: (this.presenter.approvedActionPlanSummaries.length + 1).toString(),
```

Comment:
perhaps we should remove the version number from non-approved APs to reduce potential confusion.